### DISCUSSION TITLE
For installations - git pull suggest a depth limit

### DISCUSSION THREAD

**Discussion by goatfish100** [Homebrew/discussions#74] [Everyday usage]
I've been having some repeated system issues - hence I've re installed linux brew a few times.  In the getting starting page - I believe it would make sense to limit the git pull to a certain depth.  I note a very slow download if I don't add a depth limit.

https://docs.brew.sh/Homebrew-on-Linux

I note a very fast download if I use this 
git clone --depth 30 https://github.com/Homebrew/brew ~/.linuxbrew/Homebrew

And a very slow download with no depth limit

**Comment by jonchang** [comment]
Please don't do this. This imposes a significant performance penalty on all update operations, and GitHub has asked us specifically to avoid shallow clones.

**Comment by rdp** [comment]
I wonder if homebrew could use something else besides "git clones" for updating formulae...it always takes forever when it kicks in (though, to be fair, yum updates also take forever LOL).

**Reply by carlocab** [reply]
Try`HOMEBREW_INSTALL_FROM_API`. See `man brew` for details. Note: this feature is still experimental.
